infer/environment.cc

```diff
@@ -1392,7 +1392,7 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
 
                 const core::TypeAndOrigins &typeAndOrigin = getAndFillTypeAndOrigin(ctx, i.what);
                 if (core::Types::isSubType(ctx, core::Types::void_(), methodReturnType)) {
-                    methodReturnType = core::Types::untypedUntracked();
+                    methodReturnType = core::Types::top();
```

Comment:
Do you want to land this separately? 
I don't mind landing it here too, but it seems ti be unrelated to other changes in this PR

Reply:
Do we also have tests for that?

Reply:
The problem here is that then the tests in `test/testdata/infer/strong_block_return.rb` (and the `returns_untyped` method in `test/testdata/infer/untyped.rb`) would not pass—the check below which says "if `methodReturnType` is `T.untyped`, report that there's no type coverage for the method result."

While technically you're right that I could split them up, it felt like the churn to the test cases would have been annoying enough that I wanted to include them in this change.

lmk if it's okay with you to proceed with one PR.